tools/PI/DevHome.PI/BarWindow.xaml.cs

```diff
@@ -0,0 +1,722 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using DevHome.Common.Extensions;
+using DevHome.PI.Controls;
+using DevHome.PI.Helpers;
+using DevHome.PI.Models;
+using DevHome.PI.Properties;
+using DevHome.PI.Telemetry;
+using DevHome.PI.ViewModels;
+using Microsoft.UI;
+using Microsoft.UI.Windowing;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Input;
+using Windows.UI.WindowManagement;
+using Windows.Win32;
+using Windows.Win32.Foundation;
+using Windows.Win32.UI.Accessibility;
+using Windows.Win32.UI.WindowsAndMessaging;
+using WinRT.Interop;
+using WinUIEx;
+using static DevHome.PI.Helpers.WindowHelper;
+
+namespace DevHome.PI;
+
+public partial class BarWindow : WindowEx, INotifyPropertyChanged
+{
+    private readonly Settings settings = Settings.Default;
+    private readonly string errorTitleText = CommonHelper.GetLocalizedString("ToolLaunchErrorTitle");
+    private readonly string errorMessageText = CommonHelper.GetLocalizedString("ToolLaunchErrorMessage");
+    private readonly BarWindowViewModel viewModel = new();
+
+    // Constants that control window sizes
+    private const int WindowPositionOffsetY = 30;
+    private const int FloatingHorizontalBarWidth = 700;

```

Comment:
How does this handle system text size being scaled up? There might be an accessibility bug here.